### ISSUE TITLE
Overfitting 

### ISSUE THREAD

**Issue by samshipengs** [samshipengs/Coordinated-Multi-Agent-Imitation-Learning#15]
> 2018-05-19 20:15:18,769 | INFO : Training with hyper parameters:
{'use_model': 'dynamic_rnn_layer_norm', 'batch_size': 64, 'sequence_length': 50, 'overlap': 25, 'state_size': [128, 128], 'use_peepholes': None, 'input_dim': 179, 'dropout_rate': 0.6, 'learning_rate': 0.0001, 'n_epoch': 1000}

> 2018-05-19 20:15:22,532 | INFO : Horizon 0 ==========
2018-05-19 20:15:58,637 | INFO : Epoch 0    | loss: 424.64   | time took: 34.02s | validation loss: 370.09  
2018-05-19 20:21:29,900 | INFO : Epoch 10   | loss: 13.52    | time took: 32.61s | validation loss: 10.31   
2018-05-19 20:26:59,878 | INFO : Epoch 20   | loss: 3.62     | time took: 32.94s | validation loss: 3.74    
2018-05-19 20:32:30,784 | INFO : Epoch 30   | loss: 3.08     | time took: 32.63s | validation loss: 2.86    
2018-05-19 20:37:59,144 | INFO : Epoch 40   | loss: 2.90     | time took: 32.77s | validation loss: 2.74    
2018-05-19 20:43:26,475 | INFO : Epoch 50   | loss: 2.79     | time took: 32.46s | validation loss: 2.58    
2018-05-19 20:48:53,346 | INFO : Epoch 60   | loss: 2.70     | time took: 32.50s | validation loss: 2.70    
2018-05-19 20:54:20,320 | INFO : Epoch 70   | loss: 2.63     | time took: 32.46s | validation loss: 2.56    
2018-05-19 20:59:47,275 | INFO : Epoch 80   | loss: 2.57     | time took: 32.45s | validation loss: 2.60    
2018-05-19 21:05:14,118 | INFO : Epoch 90   | loss: 2.50     | time took: 32.50s | validation loss: 2.59    
2018-05-19 21:10:43,281 | INFO : Epoch 100  | loss: 2.43     | time took: 32.98s | validation loss: 2.55    
2018-05-19 21:16:13,949 | INFO : Epoch 110  | loss: 2.37     | time took: 33.29s | validation loss: 2.57    
2018-05-19 21:21:45,966 | INFO : Epoch 120  | loss: 2.30     | time took: 32.93s | validation loss: 2.51    
2018-05-19 21:27:17,978 | INFO : Epoch 130  | loss: 2.24     | time took: 32.95s | validation loss: 2.44    
2018-05-19 21:32:49,640 | INFO : Epoch 140  | loss: 2.18     | time took: 32.94s | validation loss: 2.72    
2018-05-19 21:38:21,280 | INFO : Epoch 150  | loss: 2.12     | time took: 32.89s | validation loss: 2.48    
2018-05-19 21:43:51,814 | INFO : Epoch 160  | loss: 2.07     | time took: 32.71s | validation loss: 2.62    
2018-05-19 21:49:22,015 | INFO : Epoch 170  | loss: 2.02     | time took: 32.80s | validation loss: 2.57    
2018-05-19 21:54:51,845 | INFO : Epoch 180  | loss: 1.96     | time took: 32.79s | validation loss: 2.57    
2018-05-19 22:00:21,549 | INFO : Epoch 190  | loss: 1.91     | time took: 32.69s | validation loss: 2.62    
2018-05-19 22:05:51,438 | INFO : Epoch 200  | loss: 1.86     | time took: 32.75s | validation loss: 2.52    
2018-05-19 22:11:21,148 | INFO : Epoch 210  | loss: 1.82     | time took: 32.73s | validation loss: 2.60    
2018-05-19 22:16:50,945 | INFO : Epoch 220  | loss: 1.78     | time took: 32.74s | validation loss: 2.67    
2018-05-19 22:22:21,564 | INFO : Epoch 230  | loss: 1.74     | time took: 32.80s | validation loss: 2.65    
2018-05-19 22:27:51,485 | INFO : Epoch 240  | loss: 1.70     | time took: 32.78s | validation loss: 2.84    
2018-05-19 22:33:21,232 | INFO : Epoch 250  | loss: 1.66     | time took: 32.75s | validation loss: 2.59    
2018-05-19 22:38:51,611 | INFO : Epoch 260  | loss: 1.62     | time took: 33.05s | validation loss: 2.66    
2018-05-19 22:44:21,437 | INFO : Epoch 270  | loss: 1.59     | time took: 32.74s | validation loss: 2.68    
2018-05-19 22:49:51,152 | INFO : Epoch 280  | loss: 1.55     | time took: 32.75s | validation loss: 2.61    
2018-05-19 22:55:20,930 | INFO : Epoch 290  | loss: 1.52     | time took: 32.71s | validation loss: 2.55    
2018-05-19 23:00:50,665 | INFO : Epoch 300  | loss: 1.49     | time took: 32.75s | validation loss: 2.74    
2018-05-19 23:06:20,603 | INFO : Epoch 310  | loss: 1.46     | time took: 32.75s | validation loss: 2.80    
2018-05-19 23:11:50,676 | INFO : Epoch 320  | loss: 1.43     | time took: 32.75s | validation loss: 2.75    
2018-05-19 23:17:20,512 | INFO : Epoch 330  | loss: 1.40     | time took: 32.76s | validation loss: 2.78    
2018-05-19 23:22:50,682 | INFO : Epoch 340  | loss: 1.37     | time took: 32.76s | validation loss: 3.07    
2018-05-19 23:28:20,826 | INFO : Epoch 350  | loss: 1.34     | time took: 32.75s | validation loss: 2.82    
2018-05-19 23:33:50,696 | INFO : Epoch 360  | loss: 1.32     | time took: 32.74s | validation loss: 2.77    
2018-05-19 23:39:20,657 | INFO : Epoch 370  | loss: 1.30     | time took: 32.94s | validation loss: 2.82    
2018-05-19 23:44:50,417 | INFO : Epoch 380  | loss: 1.27     | time took: 32.84s | validation loss: 2.94    
2018-05-19 23:50:20,729 | INFO : Epoch 390  | loss: 1.24     | time took: 32.97s | validation loss: 2.79    
2018-05-19 23:55:55,456 | INFO : Epoch 400  | loss: 1.22     | time took: 33.06s | validation loss: 2.88    
2018-05-20 00:01:30,456 | INFO : Epoch 410  | loss: 1.20     | time took: 32.58s | validation loss: 2.94    
2018-05-20 00:07:07,730 | INFO : Epoch 420  | loss: 1.18     | time took: 33.55s | validation loss: 3.09    
2018-05-20 00:12:44,634 | INFO : Epoch 430  | loss: 1.16     | time took: 33.37s | validation loss: 2.78    
2018-05-20 00:18:14,081 | INFO : Epoch 440  | loss: 1.14     | time took: 32.67s | validation loss: 3.27    
2018-05-20 00:23:47,005 | INFO : Epoch 450  | loss: 1.12     | time took: 33.36s | validation loss: 2.82    
2018-05-20 00:29:12,431 | INFO : Epoch 460  | loss: 1.10     | time took: 32.09s | validation loss: 3.01    
2018-05-20 00:34:35,043 | INFO : Epoch 470  | loss: 1.08     | time took: 32.03s | validation loss: 2.63    
2018-05-20 00:39:58,028 | INFO : Epoch 480  | loss: 1.07     | time took: 32.10s | validation loss: 2.92    
2018-05-20 00:45:20,698 | INFO : Epoch 490  | loss: 1.05     | time took: 32.07s | validation loss: 3.15    
2018-05-20 00:50:43,453 | INFO : Epoch 500  | loss: 1.03     | time took: 32.09s | validation loss: 2.89    
2018-05-20 00:56:06,164 | INFO : Epoch 510  | loss: 1.01     | time took: 32.08s | validation loss: 2.87    
2018-05-20 01:01:28,880 | INFO : Epoch 520  | loss: 1.00     | time took: 32.04s | validation loss: 3.02    
2018-05-20 01:06:52,427 | INFO : Epoch 530  | loss: 0.98     | time took: 32.94s | validation loss: 2.94    
2018-05-20 01:12:15,404 | INFO : Epoch 540  | loss: 0.96     | time took: 32.09s | validation loss: 2.90    
2018-05-20 01:17:38,158 | INFO : Epoch 550  | loss: 0.96     | time took: 32.09s | validation loss: 2.80    
2018-05-20 01:23:01,047 | INFO : Epoch 560  | loss: 0.93     | time took: 32.13s | validation loss: 3.14    
2018-05-20 01:28:23,858 | INFO : Epoch 570  | loss: 0.93     | time took: 32.08s | validation loss: 2.94    
2018-05-20 01:33:46,562 | INFO : Epoch 580  | loss: 0.91     | time took: 32.07s | validation loss: 2.79    
2018-05-20 01:39:09,246 | INFO : Epoch 590  | loss: 0.90     | time took: 32.07s | validation loss: 3.16    
2018-05-20 01:44:31,918 | INFO : Epoch 600  | loss: 0.89     | time took: 32.08s | validation loss: 2.85    
2018-05-20 01:49:54,587 | INFO : Epoch 610  | loss: 0.87     | time took: 32.09s | validation loss: 2.90    
2018-05-20 01:55:17,323 | INFO : Epoch 620  | loss: 0.86     | time took: 32.03s | validation loss: 2.67    
2018-05-20 02:00:40,017 | INFO : Epoch 630  | loss: 0.85     | time took: 32.07s | validation loss: 3.18    
2018-05-20 02:06:02,918 | INFO : Epoch 640  | loss: 0.84     | time took: 32.09s | validation loss: 3.01    
2018-05-20 02:11:25,634 | INFO : Epoch 650  | loss: 0.83     | time took: 32.05s | validation loss: 2.93    
2018-05-20 02:16:48,348 | INFO : Epoch 660  | loss: 0.82     | time took: 32.06s | validation loss: 3.02    
2018-05-20 02:22:11,146 | INFO : Epoch 670  | loss: 0.81     | time took: 32.07s | validation loss: 2.79    
2018-05-20 02:27:33,876 | INFO : Epoch 680  | loss: 0.80     | time took: 32.12s | validation loss: 2.88    
2018-05-20 02:32:56,727 | INFO : Epoch 690  | loss: 0.79     | time took: 32.13s | validation loss: 2.95    
2018-05-20 02:38:19,538 | INFO : Epoch 700  | loss: 0.77     | time took: 32.07s | validation loss: 2.66    
2018-05-20 02:43:42,278 | INFO : Epoch 710  | loss: 0.76     | time took: 32.05s | validation loss: 2.89    
2018-05-20 02:49:05,039 | INFO : Epoch 720  | loss: 0.76     | time took: 32.11s | validation loss: 2.81    
2018-05-20 02:54:27,839 | INFO : Epoch 730  | loss: 0.75     | time took: 32.08s | validation loss: 2.98    
2018-05-20 02:59:50,629 | INFO : Epoch 740  | loss: 0.74     | time took: 32.08s | validation loss: 2.97    
2018-05-20 03:05:13,247 | INFO : Epoch 750  | loss: 0.73     | time took: 32.07s | validation loss: 2.82    
2018-05-20 03:10:35,949 | INFO : Epoch 760  | loss: 0.72     | time took: 32.04s | validation loss: 2.81    
2018-05-20 03:15:58,732 | INFO : Epoch 770  | loss: 0.72     | time took: 32.07s | validation loss: 2.86    
2018-05-20 03:21:21,429 | INFO : Epoch 780  | loss: 0.70     | time took: 32.07s | validation loss: 2.81    
2018-05-20 03:26:44,166 | INFO : Epoch 790  | loss: 0.70     | time took: 32.07s | validation loss: 2.73    
2018-05-20 03:32:06,811 | INFO : Epoch 800  | loss: 0.69     | time took: 32.07s | validation loss: 2.93    
2018-05-20 03:37:29,600 | INFO : Epoch 810  | loss: 0.68     | time took: 32.13s | validation loss: 3.20    
2018-05-20 03:42:52,533 | INFO : Epoch 820  | loss: 0.67     | time took: 32.09s | validation loss: 2.85    
2018-05-20 03:48:15,239 | INFO : Epoch 830  | loss: 0.66     | time took: 32.05s | validation loss: 2.91    
2018-05-20 03:53:38,068 | INFO : Epoch 840  | loss: 0.66     | time took: 32.11s | validation loss: 2.95    
2018-05-20 03:59:00,843 | INFO : Epoch 850  | loss: 0.65     | time took: 32.06s | validation loss: 2.93    
2018-05-20 04:04:23,524 | INFO : Epoch 860  | loss: 0.64     | time took: 32.05s | validation loss: 3.17    
2018-05-20 04:09:46,182 | INFO : Epoch 870  | loss: 0.64     | time took: 32.06s | validation loss: 2.82    
2018-05-20 04:15:08,897 | INFO : Epoch 880  | loss: 0.63     | time took: 32.09s | validation loss: 2.85    
2018-05-20 04:20:31,599 | INFO : Epoch 890  | loss: 0.63     | time took: 32.07s | validation loss: 2.86    
2018-05-20 04:25:54,164 | INFO : Epoch 900  | loss: 0.62     | time took: 32.08s | validation loss: 2.72    
2018-05-20 04:31:16,932 | INFO : Epoch 910  | loss: 0.61     | time took: 32.08s | validation loss: 2.88    
2018-05-20 04:36:39,548 | INFO : Epoch 920  | loss: 0.61     | time took: 32.05s | validation loss: 2.95    
2018-05-20 04:42:02,369 | INFO : Epoch 930  | loss: 0.60     | time took: 32.03s | validation loss: 2.87    
2018-05-20 04:47:25,195 | INFO : Epoch 940  | loss: 0.60     | time took: 32.07s | validation loss: 2.83    
2018-05-20 04:52:47,975 | INFO : Epoch 950  | loss: 0.59     | time took: 32.13s | validation loss: 2.86    
2018-05-20 04:58:10,736 | INFO : Epoch 960  | loss: 0.59     | time took: 32.06s | validation loss: 3.23    
2018-05-20 05:03:33,409 | INFO : Epoch 970  | loss: 0.58     | time took: 32.09s | validation loss: 3.18    
2018-05-20 05:08:56,194 | INFO : Epoch 980  | loss: 0.58     | time took: 32.10s | validation loss: 3.03    
2018-05-20 05:14:18,889 | INFO : Epoch 990  | loss: 0.57     | time took: 32.08s | validation loss: 2.96    
2018-05-20 05:19:07,667 | INFO : Total time took: 9.06hrs
2018-05-20 05:19:07,824 | INFO : Done saving model for policy 0
